### PR TITLE
Update README for YAML Rest API specs

### PR DESCRIPTION
**Pull request by mosche**
This documents the recent changes made to the YAML specs dsl in the `skip` and the new `requires` section.

### CODE REVIEW

File: rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/README.asciidoc [hunk 1]
```diff
@@ -87,7 +87,106 @@ in the `indices.get_settings` API.
...
+
+A `requires` section must specify at least one of the two options above.
+If `cluster_features` are required, a `reason` must be provided in addition.
+
+A `skip` section, on the other hand, defines certain conditions that, if met, will skip the test, such as:
+
+- `cluster_features` to <<cluster_features, skip if cluster features are present>>.
+- `known_issues` to <<skip_known_issues, skip on known issues (based on cluster features)>>.
+- `awaits_fix` to <<skip_awaits_fix, always skip / mute a test due to a pending fix>>.
+- `os` to <<skip_os, skip on operating systems>>.
```
Comment (thecoop) [inline rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/README.asciidoc:107]: What's the << syntax? I don't quite understand the 'to'

File: rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/README.asciidoc [hunk 2]
```diff
@@ -87,7 +87,106 @@ in the `indices.get_settings` API.
...
+`known_issues` provides a more explicit way to express and skip a certain range of buggy releases based on cluster features.
+Each of possibly multiple issues is a pair of `cluster_feature` and `fixed_by`, where an issue was
+introduced by the former feature and eventually fixed by the latter one. For instance:
+
+....
+    "Parent":
+     - skip:
+          known_issues:
+            - cluster_feature: feature_y
+              fixed_by:        feature_y_fix
```
Comment (thecoop) [inline rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/README.asciidoc:162]: May be worth making clear these can be version features too

File: rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/README.asciidoc [hunk 3]
```diff
@@ -87,7 +87,106 @@ in the `indices.get_settings` API.
...
+
+Previously, it was possible to skip ranges of broken release versions using <<skip_version,`version`>>.
+`known_issues` provides a more explicit way to express and skip a certain range of buggy releases based on cluster features.
+Each of possibly multiple issues is a pair of `cluster_feature` and `fixed_by`, where an issue was
+introduced by the former feature and eventually fixed by the latter one. For instance:
+
+....
+    "Parent":
+     - skip:
+          known_issues:
```
[the remark this answers is not in the document]
Comment (mosche) [reply rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/README.asciidoc:160]: `known_issues` contains a list, I've added a 2nd one to clarify

File: rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/README.asciidoc (hunk 2 above)
Comment (mosche) [reply rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/README.asciidoc:162]: I'm mentioning that just below the example 👍

**Review by thecoop — APPROVED** [review]

**Comment by mosche** [comment]
@elasticmachine update branch

**Review by ldematte — APPROVED** [review]
👍 
one very minor nit, include it if you want. Otherwise, good as it is!

**Pull request merged**